pkg/middleware/consistency/consistency.go

```diff
@@ -232,24 +269,48 @@ func (s *recvWrapper) RecvMsg(m interface{}) error {
 		return err
 	}
 	ds := datastoremw.MustFromContext(s.ctx)
-	return s.handler(s.ctx, m, ds, s.serviceLabel)
+	return s.handler(s.ctx, m, ds, s.serviceLabel, s.option)
 }
 
 // pickBestRevision compares the provided ZedToken with the optimized revision of the datastore, and returns the most
 // recent one. The boolean return value will be true if the provided ZedToken is the most recent, false otherwise.
-func pickBestRevision(ctx context.Context, requested *v1.ZedToken, ds datastore.Datastore) (datastore.Revision, bool, error) {
+func pickBestRevision(ctx context.Context, requested *v1.ZedToken, ds datastore.Datastore, option MismatchingTokenOption) (datastore.Revision, bool, error) {
 	// Calculate a revision as we see fit
 	databaseRev, err := ds.OptimizedRevision(ctx)
 	if err != nil {
 		return datastore.NoRevision, false, err
 	}
 
 	if requested != nil {
-		requestedRev, err := zedtoken.DecodeRevision(requested, ds)
+		requestedRev, status, err := zedtoken.DecodeRevision(requested, ds)
 		if err != nil {
 			return datastore.NoRevision, false, errInvalidZedToken
 		}
 
+		if status == zedtoken.StatusMismatchedDatastoreID {
+			switch option {
+			case TreatMismatchingTokensAsFullConsistency:
+				log.Warn().Str("zedtoken", requested.Token).Msg("ZedToken specified references a different datastore instance and SpiceDB is configured to treat this as a full consistency request")
```

Comment:
non-blocking comment: assuming that spicedb logs all request parameters, including `consistency`, would it not be more useful to add a new field `overriden-consistency` or something like that? otherwise when you are going through logs looking for a particular request, you will not get the full picture just by looking at one log line; you will need to look at some lines before and after